### PR TITLE
Archive Orders Stored Procedure (#1278)

### PR DESCRIPTION
**Pull request by ritesh-skyach**
Archive Orders Stored Procedure

### CODE REVIEW

File: db/migrate/20180716110346_add_archive_orders_stored_procedure.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+class AddArchiveOrdersStoredProcedure < ActiveRecord::Migration
+  def self.up
+    execute 'DROP procedure IF EXISTS `archive_orders`;'
+    execute <<-SQL
+    CREATE PROCEDURE archive_orders ()
+    BEGIN
+    SET @prepare_table = CONCAT('CREATE TABLE IF NOT EXISTS ', @table_name, ' SELECT * FROM orders WHERE 1=0');
+    PREPARE create_table from @prepare_table;
```
Comment (mod) [inline db/migrate/20180716110346_add_archive_orders_stored_procedure.rb]: @ritesh-skyach why using PREPARE and Execute statement,
the Stored procedure will be much more readable with simple statements.

a simple

CREATE TABLE `orders_201806` IF NOT EXIST
INSERT FROM SELECT `orders`
DELETE FROM `orders`

**Review by mod — CHANGES_REQUESTED** [review]

File: db/migrate/20180716110346_add_archive_orders_stored_procedure.rb (hunk 1 above)
Comment (ritesh-skyach) [reply db/migrate/20180716110346_add_archive_orders_stored_procedure.rb]: @mod We can not use variable name in SQL statement directly, So we have to use PREPARE and EXECUTE

**Comment by mod** [comment]
I still think you can completely avoid the prepare statements, why do you user them ?

**Comment by ritesh-skyach** [comment]
@mod We are using dynamic table name so it is not possible to use without prepare and execute.
Please correct me if i am wrong.

**Review by mod — CHANGES_REQUESTED** [review]
@ritesh-skyach Please update this SP with CREATE table statement with no PREPARE statements

**Pull request closed without merging**